examples/private_connection/main.tf

```diff
@@ -0,0 +1,190 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+provider "google" {
+  version = "~> 3.7.0"
+  project = var.project_id
+}
+
+locals {
+  service_list = [
+    "compute.googleapis.com",
+    "cloudresourcemanager.googleapis.com",
+    "serviceusage.googleapis.com",
+    "servicenetworking.googleapis.com",
+    "dns.googleapis.com"
+  ]
+}
+
+resource "google_project_service" "main" {
+  count              = length(local.service_list)
+  project            = var.project_id
+  service            = local.service_list[count.index]
+  disable_on_destroy = "false"
+}
+
+resource "google_compute_network" "network" {
```

Comment:
Other examples assume the network/subnetwork exists already. So maybe follow their convention to consume the network/subnetwork as an inputs. @gkowalski-google WDYT?

Reply:
the problem is that existing network should have `private_ip_google_access = true`. what is a best way to enforce it?

Reply:
i can configure network and subnet as input vars and validate that the subnet has private google access enabled. if it doesn't, i will trigger an error. is it OK?

Reply:
I'm ok either way. Creating the network and subnet is also fine with me.

Reply:
then let's leave it "as is" for now.